ui/src/org/pentaho/di/ui/core/database/wizard/CreateDatabaseWizardPageJDBC.java

```diff
@@ -163,39 +168,58 @@ public void modifyText( ModifyEvent arg0 ) {
 
   public void setData() {
     wHostname.setText( Const.NVL( databaseMeta.getHostname(), "" ) );
-
     wPort.setText( databaseMeta.getDatabasePortNumberString() );
 
-    wDBName.setText( Const.NVL( databaseMeta.getDatabaseName(), "" ) );
+    if ( !defaultWebAppNameSet && DATA_SERVICES_PLUGIN_ID.equals( databaseMeta.getPluginId() ) ) {
+      wDBName.setText( DEFAULT_WEB_APPLICATION_NAME );
+      defaultWebAppNameSet = true;
+    } else {
+      wDBName.setText( Const.NVL( databaseMeta.getDatabaseName(), "" ) );
+    }
+
+    if ( DATA_SERVICES_PLUGIN_ID.equals( databaseMeta.getPluginId() ) ) {
+      wlDBName.setText( BaseMessages.getString( PKG, "CreateDatabaseWizardPageJDBC.WebAppName.Label" ) );
+    } else {
+      wlDBName.setText( BaseMessages.getString( PKG, "CreateDatabaseWizardPageJDBC.DBName.Label" ) );
+    }
   }
 
   public boolean canFlipToNextPage() {
     String server =
-      wHostname.getText() != null ? wHostname.getText().length() > 0 ? wHostname.getText() : null : null;
+        wHostname.getText() != null ? wHostname.getText().length() > 0 ? wHostname.getText() : null : null;
     String port = wPort.getText() != null ? wPort.getText().length() > 0 ? wPort.getText() : null : null;
     String dbname = wDBName.getText() != null ? wDBName.getText().length() > 0 ? wDBName.getText() : null : null;
 
-    if ( server == null || port == null || dbname == null ) {
+    if ( ( server == null || port == null || dbname == null ) && !DATA_SERVICES_PLUGIN_ID
+        .equals( databaseMeta.getPluginId() ) ) {
       setErrorMessage( BaseMessages.getString( PKG, "CreateDatabaseWizardPageJDBC.ErrorMessage.InvalidInput" ) );
+
+      return false;
+    }
+
+    if ( ( server == null || port == null ) && DATA_SERVICES_PLUGIN_ID.equals( databaseMeta.getPluginId() ) ) {
+      setErrorMessage( BaseMessages.getString( PKG, "CreateDatabaseWizardPageJDBC.PDSHostPort.ErrorMessage" ) );
+
       return false;
-    } else {
-      getDatabaseInfo();
-      setErrorMessage( null );
-      setMessage( BaseMessages.getString( PKG, "CreateDatabaseWizardPageJDBC.Message.Input" ) );
-      return true;
     }
+
+    getDatabaseInfo();
+    setErrorMessage( null );
+    setMessage( BaseMessages.getString( PKG, "CreateDatabaseWizardPageJDBC.Message.Input" ) );
+
+    return true;
   }
 
   public DatabaseMeta getDatabaseInfo() {
-    if ( wHostname.getText() != null && wHostname.getText().length() > 0 ) {
+    if ( StringUtil.isEmpty( wHostname.getText() ) ) {
       databaseMeta.setHostname( wHostname.getText() );
     }
 
-    if ( wPort.getText() != null && wPort.getText().length() > 0 ) {
+    if ( StringUtil.isEmpty( wPort.getText() ) ) {
       databaseMeta.setDBPort( wPort.getText() );
     }
 
-    if ( wDBName.getText() != null && wDBName.getText().length() > 0 ) {
+    if ( wDBName.getText() != null ) {
```

Comment:
I do not see any hard in having all 3 of them check for null only, but to minimize number of changes I didn't touch other 2 because of possible regressions/unwanted impacts and so on.